util/types/datum.go

```diff
@@ -960,40 +960,27 @@ func (d *Datum) convertToMysqlTimestamp(sc *variable.StatementContext, target *F
 			t.TimeZone = loc
 		case t.TimeZone == time.UTC:
 			// Convert to session timezone.
-			if !t.IsZero() {
-				raw, err := t.Time.GoTime(time.UTC)
-				if err != nil {
-					return ret, errors.Trace(err)
-				}
-				converted := raw.In(loc)
-				t.Time = FromGoTime(converted)
-			}
-			t.TimeZone = loc
+			t.ConvertTimeZone(time.UTC, loc)
 		case t.TimeZone == sc.TimeZone:
-			break
+		default:
+			return ret, errors.New("convertToMysqlTimestamp get a wrong input")
```

Comment:
The error information is not important, because it should never meet this error in practice, 
maybe I should use panic().